pkg/daemon/ceph/client/rados.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2022 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/util/exec"
+)
+
+/*
+Example Lock info:
+
+	{
+	  "name": "rook-ceph-nfs",
+	  "type": "exclusive",
+	  "tag": "rook-ceph-nfs",
+	  "lockers": [
+	    {
+	      "name": "client.28945",
+	      "cookie": "test-cookie",
+	      "description": "",
+	      "expiration": "2022-09-08T23:56:57.924802+0000",
+	      "addr": "10.244.0.44:0/960227889"
+	    }
+	  ]
+	}
+*/
+type radosLockInfo struct {
+	Name    string            `json:"name"`
+	Tag     string            `json:"tag"`
+	Lockers []radosLockerInfo `json:"lockers"`
+}
+
+type radosLockerInfo struct {
+	Name   string `json:"name"`
+	Cookie string `json:"cookie"`
+}
+
+// RadosLockObject locks a rados object in a given pool and namespace and returns a lock "cookie"
+// that can be used to identify this unique lock.
+func RadosLockObject(
+	context *clusterd.Context, clusterInfo *ClusterInfo,
+	pool, namespace, objectName, lockName string,
+	lockTimeout time.Duration,
+) (string, error) {
+	// generate a random "cookie" that identifies this lock "session"
+	cookieLen := 12
```

Comment:
```suggestion
	cookieLen := 12
```
is length 12 fixed? if yes, can we move it up and create `const` var?

Reply:
It's fixed but fairly arbitrary and not super relevant elsewhere. It's only used in this function and made a variable rather than a const to help better read the code.